Doc/library/urllib.request.rst

```diff
@@ -1557,9 +1549,38 @@ some point in the future.
    :synopsis: Response classes used by urllib.
 
 The :mod:`urllib.response` module defines functions and classes which define a
-minimal file like interface, including ``read()`` and ``readline()``. The
-typical response object is an addinfourl instance, which defines an ``info()``
-method and that returns headers and a ``geturl()`` method that returns the url.
-Functions defined by this module are used internally by the
-:mod:`urllib.request` module.
+minimal file-like interface, including ``read()`` and ``readline()``.
+Functions defined by this module are used internally by the :mod:`urllib.request` module.
+The typical response object is a :class:`urllib.response.addinfourl` instance:
+
+.. class:: urllib.response.addinfourl
+
+.. attribute:: addinfourl.url
```

Comment:
To avoid repeating the class name, nest the attribute/method directives under the class directive:

```rst

.. class:: addinfourl

   .. attribute:: headers
```

(adding the module name to the class is not needed because the file has a `module` directive at the top)

Reply:
Got it, thanks.